crates/catalog/rest/src/catalog.rs

```diff
@@ -497,13 +516,56 @@ impl RestCatalog {
             client: config.try_create_rest_client()?,
             config,
         };
-
+        catalog.fetch_access_token().await?;
+        catalog.client = catalog.config.try_create_rest_client()?;
         catalog.update_config().await?;
         catalog.client = catalog.config.try_create_rest_client()?;
 
         Ok(catalog)
     }
 
+    async fn fetch_access_token(&mut self) -> Result<()> {
+        if self.config.props.contains_key("token") {
+            return Ok(());
+        }
```

Comment:
I don't think we should add this check since toke may expire, and we need to refresh it.

Reply:
Sorry, I still don't get your point. But currently we in fact don't refresh tokens, so this logic is same.

Reply:
Anyway, the logic is same as a specified python version, I think it's acceptable.